cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -579,11 +581,26 @@ rebuildTargets verbosity
       (Set.toList . Set.fromList)
         [ pkgdb
         | InstallPlan.Configured elab <- InstallPlan.toList installPlan
-        , (pkgdb:_) <- map reverse [ elabBuildPackageDBStack elab,
-                                     elabRegisterPackageDBStack elab,
-                                     elabSetupPackageDBStack elab ]
+        , pkgdb <- concat [ elabBuildPackageDBStack elab
```

Comment:
This has nothing to do with the current patch, but I wonder, should we really be in the business of creating the global package database (which I imagine is recorded in the database stack?) NB: this list comprension is being used to compute packageDBsToUse, the list of package dbs we may need to create.